core/tests/puppeteer-acceptance-tests/utilities/user/exploration-editor.ts

```diff
@@ -1122,6 +1136,133 @@ export class ExplorationEditor extends BaseUser {
+   * @param {boolean} isPresent - Whether the misconception is expected to be present.
+   */
+  async verifyMisconceptionPresentForState(
+    misconceptionName: string,
+    isPresent: boolean
+  ): Promise<void> {
+    try {
+      await this.page.waitForSelector(misconceptionDiv, {
+        timeout: 5000,
+        visible: true,
+      });
+      const misconceptions = await this.page.$$(misconceptionDiv);
+
+      for (const misconception of misconceptions) {
+        const title = await this.page.evaluate(
+          el => el.textContent,
+          misconception
+        );
+        if (title.trim() === misconceptionName) {
+          if (!isPresent) {
+            throw new Error(
+              `The misconception ${misconceptionName} is present, which was not expected`
+            );
+          }
+          return;
+        }
+      }
+
+      if (isPresent) {
+        throw new Error(
+          `The misconception ${misconceptionName} is not present, which was expected`
+        );
+      }
+    } catch (error) {
+      if (isPresent) {
+        throw new Error(
+          `The misconception ${misconceptionName} is not present, which was expected`
+        );
+      }
+    }
+
+    showMessage(
+      `The misconception is ${isPresent ? '' : 'not'} present as expected.`
+    );
+  }
+
+  async toggleMisconceptionApplicableStatus(
+    misconceptionName: string
+  ): Promise<void> {
+    await this.page.waitForSelector(optionalMisconceptionDiv, {
+      timeout: 5000,
+      visible: true,
+    });
+    let misconceptions = await this.page.$$(optionalMisconceptionDiv);
+    let misconceptionFound = false;
+    for (const misconception of misconceptions) {
+      const optionalMisconceptionName = await misconception.evaluate(el =>
+        el.textContent?.trim()
+      );
+      if (optionalMisconceptionName?.startsWith(misconceptionName)) {
+        const misconceptionOptions = await misconception.$(
+          optionalMisconceptionOptionsButton
+        );
+        if (!misconceptionOptions) {
+          throw new Error(
+            `Options not found for misconception "${misconceptionName}"`
+          );
+        }
+        await misconceptionOptions.click();
+        await this.page.waitForSelector(misconceptionApplicableToggle, {
+          visible: true,
+        });
+        await this.clickOn(misconceptionApplicableToggle);
+        misconceptionFound = true;
+        break;
+      }
+    }
+    if (!misconceptionFound) {
+      throw new Error(
+        `Couldn't find misconception with name ${misconceptionName}.`
+      );
+    }
+  }
+
+  async verifyOptionalMisconceptionApplicableStatus(
+    misconceptionName: string,
+    isApplicable: boolean
+  ): Promise<void> {
+    await this.verifyMisconceptionPresentForState(misconceptionName, true);
+    const inapplicableMisconceptions = await this.page.$$(
+      inapplicableMisconceptionDiv
+    );
+
+    for (const misconception of inapplicableMisconceptions) {
```

Comment:
nit: inapplicableMisconception of inapplicableMisconceptions

Reply:
I think that's confusing since they'd only differ by a letter, harder to read.